packages/terra-form-input/src/InputField.jsx

```diff
@@ -163,6 +169,7 @@ const InputField = (props) => {
       <Input
         {...inputAttrs}
         id={inputId}
+        type={type}
```

Comment:
This is a non-passive change, methinks. For the existing users who have the current code something on the lines of snippet below, this change breaks the functionality because the default value of `undefined` overrides the existing value that is set. 

```
<InputField
  inputId="numeric-input"
  label="Numeric Value"
  help="Note: This can not be changed in the future"
  inputAttrs={{
    placeholder: 'Enter Digits',
    name: 'numeric',
    type: 'number'
  }}
/>
```

Reply:
Added check for type prop : [`const inputType = (type) || inputAttrs.type;`](https://github.com/cerner/terra-core/blob/e1f32ffcf116761ffddcc01deb5d3545ab8febca/packages/terra-form-input/src/InputField.jsx#L152) . if type is not having any value assigned then the value of inputAttrs type will be assigned as type value for input.